v2/recipes/state-machine-transitions.md

```diff
@@ -0,0 +1,142 @@
+# State Machine Transitions
+
```

Comment:
Can we have an intro that also brings up the one or two spots we listen for this in Drupal Commerce? Reference http://cgit.drupalcode.org/commerce/commit/?id=890d5e2